sdk/cosmos/azure-cosmos/azure/cosmos/_session_token_helpers.py

```diff
@@ -0,0 +1,224 @@
+from ._change_feed.feed_range_internal import FeedRangeInternalEpk
+
+# pylint: disable=protected-access
+
+
+# ex inputs:
+# 1. "1:1#51", "1:1#55" -> "1:1#55"
+# 2. "0:1#57", "1:1#52" -> "0:1#57"
+def merge_session_tokens_with_same_range(session_token1: str, session_token2: str) -> str:
+    pk_range_id1, vector_session_token1 = parse_session_token(session_token1)
+    pk_range_id2, vector_session_token2 = parse_session_token(session_token2)
+    pk_range_id = pk_range_id1
+    # The partition key range id could be different in this scenario
+    # Ex. get_updated_session_token([("AA", "BB"), "1:1#51"], ("AA", "DD")) -> "1:1#51"
+    # Then we input this back into get_updated_session_token after a merge happened
+    # get_updated_session_token([("AA", "DD"), "1:1#51", ("AA", "DD"), "0:1#55"], ("AA", "DD")) -> "0:1#55"
+    if pk_range_id1 != pk_range_id2:
+        pk_range_id = pk_range_id1 \
+            if vector_session_token1.global_lsn > vector_session_token2.global_lsn else pk_range_id2
+    vector_session_token = vector_session_token1.merge(vector_session_token2)
+    return pk_range_id + ":" +  vector_session_token.session_token
+
+def is_compound_session_token(session_token: str) -> bool:
+    return "," in session_token
+
+def parse_session_token(session_token: str) -> Tuple[str, VectorSessionToken]:
+    tokens = session_token.split(":")
+    return tokens[0], VectorSessionToken.create(tokens[1])
+
+def split_compound_session_tokens(compound_session_tokens: List[Tuple[Range, str]]) -> List[str]:
+    session_tokens = []
+    for _, session_token in compound_session_tokens:
+        if is_compound_session_token(session_token):
+            tokens = session_token.split(",")
+            for token in tokens:
+                session_tokens.append(token)
+        else:
+            session_tokens.append(session_token)
+    return session_tokens
+
+# ex inputs:
+# ["1:1#51", "1:1#55", "1:1#57"] -> ["1:1#57"]
+def merge_session_tokens_for_same_partition(session_tokens: List[str]) -> List[str]:
+    i = 0
+    while i < len(session_tokens):
+        j = i + 1
+        while j < len(session_tokens):
+            pk_range_id1, vector_session_token1 = parse_session_token(session_tokens[i])
+            pk_range_id2, vector_session_token2 = parse_session_token(session_tokens[j])
+            if pk_range_id1 == pk_range_id2:
+                vector_session_token = vector_session_token1.merge(vector_session_token2)
+                session_tokens.append(pk_range_id1 + ":" + vector_session_token.session_token)
+                remove_session_tokens = [session_tokens[i], session_tokens[j]]
+                for token in remove_session_tokens:
+                    session_tokens.remove(token)
+                i = -1
+                break
+            j += 1
+        i += 1
+
+    return session_tokens
+
+# ex inputs:
+# 1. [(("AA", "BB"), "1:1#51"), (("BB", "DD"), "2:1#51"), (("AA", "DD"), "0:1#55")] ->
```

Comment:
This looks like a partition merge scenario - I think the `pkRangeId` increases? Good to call out which example is _split_ or _merge_.

Reply:
Didn't know it also increased with merge. Changed the example and added which is split and merge.